app/models/order_item.rb

```diff
@@ -16,7 +16,9 @@ class OrderItem < ApplicationRecord
   belongs_to :portfolio_item
   has_many :progress_messages, :dependent => :destroy
   has_many :approval_requests, :dependent => :destroy
+
   before_create :set_defaults
+  before_save :sanitized_parameters, :if => :will_save_change_to_service_parameters?
```

Comment:
I'm not seeing the `will_save_change_to_service_parameters?` method, but I assume/hope it's something like this:
```ruby 
def will_save_change_to_service_paremeters?
  changes.key?(:service_parameters)
end
```

also, what would you think about calling the callback method `sanitize_parameters` since it's more doing-something rather than returning something?

Reply:
@lindgrenj6 i was following the suggestion here for that `if` condition: https://stackoverflow.com/questions/24314584/run-a-callback-only-if-an-attribute-has-changed-in-rails

Reply:
Naming is hard, will definitely change that.